server/middleware/token.js

```diff
@@ -32,6 +32,9 @@ function rewriteUserLiteral(req, currentUserLiteral) {
       debug('req.url has been rewritten from %s to %s', urlBeforeRewrite,
         req.url);
     }
+  } else if (!req.accessToken && currentUserLiteral &&
```

Comment:
Love it, thought about a guard clause there, but didn't want to introduce seemingly unnecessary changes. Will update!